### PR TITLE
refactor(retrofit2): making clouddriver API call to retrofit2 from re…

### PR DESCRIPTION
**Pull request by Luthan95**
Making Changes to clouddriver API call to use retrofit2 package instead of retrofit to achieve same flow by  making use of custom calladaptor created in kork-retrofit module

### CODE REVIEW

File: rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/ArtifactDownloaderImpl.java [hunk 1]
```diff
@@ -26,12 +28,23 @@ public ArtifactDownloaderImpl(ClouddriverService clouddriverService) {
...
-    Response response =
-        retrySupport.retry(() -> clouddriverService.fetchArtifact(artifact), 5, 1000, true);
-    if (response.getBody() == null) {
+
+    ResponseBody response =
+        retrySupport.retry(
+            () -> {
+              try {
+                return clouddriverService.fetchArtifact(artifact).execute().body();
+              } catch (IOException e) {
```
Comment (dbyron-sf) [inline rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/ArtifactDownloaderImpl.java:37]: Let's talk about this.  It would be ideal to handle this more centrally instead of littering every call with a catch block like this.

File: rosco-web/src/main/groovy/com/netflix/spinnaker/rosco/Main.groovy [hunk 2]
```diff
@@ -79,7 +79,7 @@ class Main extends SpringBootServletInitializer {
     'netflix.account': '${netflix.environment}',
     'bakeAccount': '${netflix.account}',
     'netflix.stack': 'test',
-    'spring.config.additional-location': '${user.home}/.spinnaker/',
+    'spring.config.additional-location': 'optional:${user.home}/.spinnaker/',
```
Comment (dbyron-sf) [inline rosco-web/src/main/groovy/com/netflix/spinnaker/rosco/Main.groovy:82]: Let's do this in a separate PR.

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/services/ClouddriverService.java [hunk 3]
```diff
@@ -3,17 +3,24 @@
 import com.netflix.spinnaker.kork.artifacts.model.Artifact;
 import com.netflix.spinnaker.rosco.providers.aws.config.RoscoAWSConfiguration.AWSNamedImage;
 import java.util.List;
-import retrofit.client.Response;
-import retrofit.http.Body;
-import retrofit.http.GET;
-import retrofit.http.PUT;
-import retrofit.http.Query;
+// import retrofit.client.Response;
```
Comment (dbyron-sf) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/services/ClouddriverService.java:6]: Please removed the commented-out code.

File: rosco-core/rosco-core.gradle [hunk 4]
```diff
@@ -13,6 +13,9 @@ dependencies {
   implementation "io.spinnaker.kork:kork-web"
   implementation "com.squareup.retrofit:converter-jackson"
   implementation "com.squareup.retrofit:retrofit"
```
Comment (dbyron-sf) [inline rosco-core/rosco-core.gradle:15]: Can this line (and perhaps the previous one) disappear?

**Comment by Luthan95** [comment]
Duplicate and older PR Closing this
